src/mgr/PyModule.cc

```diff
@@ -43,13 +43,15 @@ std::string handle_pyerror()
     PyObject *exc, *val, *tb;
     object formatted_list, formatted;
     PyErr_Fetch(&exc, &val, &tb);
+    PyErr_NormalizeException(&exc, &val, &tb);
     handle<> hexc(exc), hval(allow_null(val)), htb(allow_null(tb));
     object traceback(import("traceback"));
     if (!tb) {
         object format_exception_only(traceback.attr("format_exception_only"));
         try {
           formatted_list = format_exception_only(hexc, hval);
         } catch (error_already_set const &) {
+          dout(1) << "Error processing exception object: " << peek_pyerror() << dendl;
```

Comment:
i'd suggest avoid printing log messages in a function which is used to compose log messages. because there are a handful callers of this function which are using `handle_pyerror()` like
```c++
derr << handle_pyerror() << dendl;
```
and i checked all callers, almost all of them use `peek_pyerror()` for composing log messages. so, in other words, the exception will always be printed in the log messages.

Reply:
I put this here, because in this specific context, `peek_pyerror()` will pick up whatever error was thrown by `format_exception()`, i.e. it's one level deeper at that point, and will tell us if format_exception() failed somehow.

Reply:
@tserong yeah, i understand. but my concern still exists. have you tried something like
```c++

derr << func() << dendl;

string func() {
  dout(1) << "oh" << dendl;
}
```
?

Reply:
Oh, sorry, now I get it. In that case it should log "oh" first, then later whatever string func() returns.

This particular error case Should Never Happen(TM) anymore, as it's only in the case when format_exception() throws an exception... And if that does happen I'd really like to know why. What about if I drop them down to log level 10 or 20? If we ever see an exception with no traceback, we'll know it's hit this case, and could increase the log level, then reproduce, to see the message.

Reply:
What about putting the error message in the string return value (where the traceback would have been)?

Reply:
That's an even better idea. Will do.